### PR TITLE
Fix erisc L1/Stack Corruption

### PR DESCRIPTION
**Pull request by tt-aho**

### CODE REVIEW

File: tt_metal/hw/inc/wormhole/eth_l1_address_map.h [hunk 1]
```diff
@@ -67,11 +52,29 @@ struct address_map {
...
   static constexpr uint32_t COMPLETION_CQ_CB_BASE = ISSUE_CQ_CB_BASE + 7 * L1_ALIGNMENT;
 
-  static constexpr std::int32_t ERISC_L1_ARG_BASE = COMPLETION_CQ_CB_BASE + 7 * L1_ALIGNMENT;
+  // Extra 4 bytes is to make sure that the launch message is properly aligned.
+  static constexpr std::int32_t ERISC_MEM_MAILBOX_BASE = COMPLETION_CQ_CB_BASE + 7 * L1_ALIGNMENT + 4;
+  // erisc early exit functionality re-uses mailboxes_t::ncrisc_halt_msg_t::stack_save memory
+  static constexpr std::int32_t ERISC_MEM_MAILBOX_STACK_SAVE = ERISC_MEM_MAILBOX_BASE + 4;
+
+  static constexpr std::int32_t ERISC_RING_BUFFER_ADDR = ERISC_MEM_MAILBOX_BASE + 128 + 12;
+  static constexpr std::int32_t PRINT_BUFFER_ER = ERISC_RING_BUFFER_ADDR + 128;
```
Comment (aliuTT) [inline tt_metal/hw/inc/wormhole/eth_l1_address_map.h:61]: Are these alignment increments? Or if they are sizes, try to grab them from `common_runtime_address_map.h`

**Review by aliuTT — APPROVED** [review]

File: tt_metal/hw/inc/wormhole/eth_l1_address_map.h (hunk 1 above)
Comment (tt-aho) [reply tt_metal/hw/inc/wormhole/eth_l1_address_map.h:61]: All of these hardcoded values/offsets are basically copied from what they were before I moved them, so I haven't really added anything new/changed anything. I'll take a look if they were defined anywhere.

File: tt_metal/hw/inc/debug/dprint.h [hunk 2]
```diff
@@ -256,7 +256,7 @@ void debug_print(DebugPrinter &dp, DebugPrintData data) {
 template<typename T>
 __attribute__((__noinline__))
 DebugPrinter operator <<(DebugPrinter dp, T val) {
-#if defined(DEBUG_PRINT_ENABLED) && !defined(PROFILE_KERNEL)
```
Comment (mo-tenstorrent) [inline tt_metal/hw/inc/debug/dprint.h:259]: This was all you had to do to get dprint and profiler working at the same time?

**Review by mo-tenstorrent — APPROVED** [review]

**Review by pgkeller — APPROVED** [review]
Please verify (either with this commit or in a follow-on commit) that the right registers are saved/restored in the assembly.  They didn't look right to me.  That list also shows up in ncrisc-halt.S

File: tt_metal/hw/inc/debug/dprint.h (hunk 2 above)
Comment (mo-tenstorrent) [reply tt_metal/hw/inc/debug/dprint.h:259]: I see, let's drop it for now and bring in it on it own PR.

**Review by ubcheema — APPROVED** [review]

**Comment by tt-aho** [comment]
> Please verify (either with this commit or in a follow-on commit) that the right registers are saved/restored in the assembly. They didn't look right to me. That list also shows up in ncrisc-halt.S

I think they look correct. The asm code in erisc.cc look like the same x# registers as specified in ncrisc-halt.S, which corresponds to the ra and 12 s# registers which are shown in the objdump.

```
    __asm__(
    9040:	fcc10113          	addi	sp, sp, -52
    9044:	00112023          	sw	ra, 0(sp)
    9048:	00812223          	sw	s0, 4(sp)
    904c:	00912423          	sw	s1, 8(sp)
    9050:	01212623          	sw	s2, 12(sp)
    9054:	01312823          	sw	s3, 16(sp)
    9058:	01412a23          	sw	s4, 20(sp)
    905c:	01512c23          	sw	s5, 24(sp)
    9060:	01612e23          	sw	s6, 28(sp)
    9064:	03712023          	sw	s7, 32(sp)
    9068:	03812223          	sw	s8, 36(sp)
    906c:	03912423          	sw	s9, 40(sp)
    9070:	03a12623          	sw	s10, 44(sp)
    9074:	03b12823          	sw	s11, 48(sp)
    9078:	00018537          	lui	a0, 0x18
    907c:	29850513          	addi	a0, a0, 664 # 18298 <__l1_bss_end+0x8600>
    9080:	00252023          	sw	sp, 0(a0)
```

**Pull request merged**